runtime/tests/sharedclassapi/sharedClasses.c

```diff
@@ -0,0 +1,625 @@
+/*******************************************************************************
+ * Copyright IBM Corp. and others 2017
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ *******************************************************************************/
+/*
+ * A jvmti agent that tests the jvmti extension functions to delete/query
+ * shared classes caches.
+ */
+
+#include <string.h>
+#include <time.h>
+#include <stdlib.h>
+
+#include "jvmti.h"
+#include "ibmjvmti.h"
+
+#define INVALID_COM_IBM_ITERATE_SHARED_CACHES_VERSION (COM_IBM_ITERATE_SHARED_CACHES_VERSION_1 + 999)
+#define INVALID_COM_IBM_ITERATE_SHARED_CACHES_NO_FLAGS (COM_IBM_ITERATE_SHARED_CACHES_NO_FLAGS + 100)
+
+/* Arguments to pass to the destroy cache function. */
+#define CACHE_TYPE_PERSISTENT 1
+#define CACHE_TYPE_NONPERSISTENT 2
+#define CACHE_TYPE_INVALID 3
+
+/* Forward declaration of the iterator callback function. */
+static jint JNICALL validateSharedCacheInfo(jvmtiEnv *jvmti, jvmtiSharedCacheInfo *cache_info, jobject user_data);
+
+/* the jvmti extension functions that will be used. */
```

Comment:
Each sentence should start with a capital (i.e. "the" -> "The" here). Please ensure all comments do likewise.